core/src/main/scala/filodb.core/query/RangeVectorCursor.scala

```diff
@@ -0,0 +1,42 @@
+package filodb.core.query
+
+import scala.collection.Iterator
+
+import filodb.memory.format.RowReader
+
+/**
+  * Please note this is not the ideal contract of cursor.
+  * Instead, it is a stop-gap implementation that gets us ability to
+  * release resources from a query. Earlier implementation purely on
+  * Iterators didnt help us nicely with that. The expectation is that
+  * moving to this trait will help us get compile time checks
+  * that force developer to care for "closing" the cursor before
+  * completing the query
+  */
+trait RangeVectorCursor extends Iterator[RowReader] { self =>
+  /**
+    * This method mut release all resources (example locks) acquired
+    * for the purpose of executing this query
+    */
+  def close(): Unit
+  def map2(f: RowReader => RowReader): RangeVectorCursor = new RangeVectorCursor {
```

Comment:
map2? Is this a common naming convention?

Reply:
There is already a `map` in `Iterator` but it returns an `Iterator`. Since I cannot use the same signature with different return value, I used `map2`. I'll rename to `mapRow`